consensus/update.go

```diff
@@ -108,7 +108,9 @@ func updatedInBlock(vc ValidationContext, b types.Block) (scos []types.SiacoinEl
 		}
 		for _, fcr := range txn.FileContractRevisions {
 			fce := fcr.Parent
-			fce.FileContract = fcr.Revision
+			if !revert {
+				fce.FileContract = fcr.Revision
+			}
```

Comment:
This will change the result of `merkle.FileContractLeaf`, so I suspect that this will cause `(merkle.ElementAccumulator).RevertBlock` to produce an invalid `RevertUpdate`: specifically, if you use it to update an existing element proof, then the proof won't match the original (that is, applying-then-reverting will result in a different proof than not-applying-in-the-first-place). Would be good to confirm this in a test.

We could get around this by leaving `FileContractLeaf` alone, and only touching the element appended to `resolved` or `revised`. I'm not totally sold on this change yet though.